server/viame_tasks/tasks.py

```diff
@@ -184,64 +184,87 @@ def run_pipeline(self: Task, params: PipelineJob):
 def train_pipeline(
     self: Task,
     results_folder: Dict,
-    source_folder: Dict,
-    groundtruth: Dict,
+    source_folder_list: List[Dict],
+    groundtruth_list: List[Dict],
     pipeline_name: str,
     config: str,
 ):
     """
     Train a pipeline by making a call to viame_train_detector
 
-    :param source_folder: The Girder Folder to pull training data from
+    :param source_folder_list: The Girder Folder to pull training data from
     :param results_folder: The Girder Folder to place the results of training into
-    :param groundtruth: The relative path to either the file containing detections,
+    :param groundtruth_list: The relative path to either the file containing detections,
         or the folder containing that file.
     :param pipeline_name: The base name of the resulting pipeline.
     """
     conf = Config()
     gc: GirderClient = self.girder_client
     manager: JobManager = self.job_manager
 
-    # Generator of items
-    training_data = gc.listItem(source_folder["_id"])
-
     viame_install_path = Path(conf.viame_install_path)
     pipeline_base_path = Path(conf.pipeline_base_path)
     training_executable = viame_install_path / "bin" / "viame_train_detector"
     config_file = pipeline_base_path / config
 
     pipeline_name = pipeline_name.replace(" ", "_")
 
+    if len(source_folder_list) != len(groundtruth_list):
+        raise Exception("Ground truth doesn't exist for all folders")
+
+    trained_on_list = []
+    input_groundtruth_list = []
     # root_data_dir is the directory passed to `viame_train_detector`
     with tempfile.TemporaryDirectory() as _temp_dir_string:
         manager.updateStatus(JobStatus.FETCHING_INPUT)
         root_data_dir = Path(_temp_dir_string)
-        download_path = Path(tempfile.mkdtemp(dir=root_data_dir))
 
-        # Download data onto server
-        gc.downloadItem(str(groundtruth["_id"]), download_path)
-        for item in training_data:
-            gc.downloadItem(str(item["_id"]), download_path)
+        # Generator of items
+        for index in range(len(source_folder_list)):
+            source_folder = source_folder_list[index]
+            groundtruth = groundtruth_list[index]
+            download_path = Path(tempfile.mkdtemp(dir=root_data_dir))
+            training_data = gc.listItem(source_folder["_id"])
+            trained_on_list.append(str(source_folder["_id"]))
+
```

Comment:
Since the for loop isn't actually a generator but `gc.listItem` is:
```suggestion
        for index in range(len(source_folder_list)):
            source_folder = source_folder_list[index]
            groundtruth = groundtruth_list[index]
            download_path = Path(tempfile.mkdtemp(dir=root_data_dir)
            trained_on_list.append(str(source_folder["_id"]))

            # Generator of items
            training_data = gc.listItem(source_folder["_id"])

```